lib/src/Uncertainty/Algorithm/Simulation/DirectionalSampling.cxx

```diff
@@ -144,8 +146,46 @@ Scalar DirectionalSampling::computeMeanContribution(const ScalarCollection & roo
 /* Compute the contribution of a set of directions direction to the probability */
 Scalar DirectionalSampling::computeTotalContribution(const Sample & directionSample)
 {
-  const UnsignedInteger sampleSize = directionSample.getSize();
   const UnsignedInteger dimension = directionSample.getDimension();
+  // First check if the value at the origin is stable wrt the event
+  // This computation has to be done only once, so we use the exception
+  // to detect the first call to getOriginValue
+  Scalar originValue;
+  try
+    {
+      // Doesn't throw an exception if already computed
+      originValue = rootStrategy_.getOriginValue();
+    }
+  catch (const NotDefinedException &)
+    {
+      // Compute the value at the origin
+      originValue = standardFunction_(Point(dimension))[0];
+      const Bool inEvent = standardEvent_.getDomain().contains(Point(1, originValue));
+      // Check if the origin is stable wrt the value at the origin
```

Comment:
@sofianehaddad originValue is used as a Scalar in several other places so it looked more convenient to store it as a Scalar in order to avoid the introduction of a new variable. Reading the code with a fresh mind, I implemented your suggestion and made other minor cleaning, thanks.